### PR TITLE
[profiling] rename profiling to internal_profiling to avoid confusion

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Renames the `profiling` option to `internal_profiling` to avoid confusion as this does not pertain to the enabling of the profiling product, but rather to the internal profiling of our agent processes. We just aim to limit confusion. This is mostly a troubleshooting feature.  

### Motivation

Avoid customer confusion.

### Additional Notes

None

### Describe your test plan

Make sure the new option applies correctly everywhere. Also, we will need to update the environment variables used in our deployments to enable/disable this correctly in our infrastructure.

### CODE REVIEW

**Review by brycekahle — APPROVED** [review]
👍

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -620,18 +620,18 @@ api_key:
 
 {{- if .Profiling -}}
 ## @param profiling - custom object - optional
-## Enter specific configurations for profiling.
+## Enter specific configurations for internal profiling.
```
Comment (robertjli) [inline pkg/config/config_template.yaml:623]: Maybe add an explicit note that this feature is used by Datadog to profile the agent itself during troubleshooting?

**Review by robertjli — APPROVED** [review]
🎉  lgtm, just left a comment to really try to make sure customers don't use this

**Review by hush-hush — APPROVED** [review]

File: pkg/config/config_template.yaml (hunk 1 above)
Comment (olivielpeau) [reply pkg/config/config_template.yaml:623]: agreed, this is worth a long explicit note (and maybe removing the link to the profiling doc URL https://docs.datadoghq.com/tracing/profiling/ as well). Can be addressed later in a docs-only PR.

**Pull request merged**